src/arch/riscv/kernel/boot.c

```diff
@@ -275,14 +277,16 @@ try_init_kernel(
     it_v_reg.start = ui_v_reg.start;
     it_v_reg.end = bi_frame_vptr + BIT(PAGE_BITS);
 
-    map_kernel_window();
-
-    /* initialise the CPU */
-    init_cpu();
+    if (!config_set(CONFIG_SEL4_RV_MACHINE)) {
+        map_kernel_window();
+    }
 
     /* initialize the platform */
     init_plat(dtb_reg);
 
+    /* initialise the CPU */
+    init_cpu();
```

Comment:
It's not clear to me why moving `init_cpu` is required to not map the kernel window in M-mode